### PR TITLE
Fix parquet loader regression type

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**

### CODE REVIEW

File: modules/parquet/src/parquetjs/utils/decoders.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export async function decodeDataPages(
       continue;
     }
 
-    if (dictionary) {
+    if (dictionary.length) {
```
Comment (ibgreen) [inline modules/parquet/src/parquetjs/utils/decoders.ts:66]: Are we sure dictionary is defined? 
```suggestion
    if (dictionary?.length) {
```

Also it is still really confusing me that this file is in the `utils` directory. This file seems to contain the meat of the parquet decoding logic, not some minor utilities.

File: modules/parquet/src/parquetjs/classes/parquet-envelope-reader.ts [hunk 2]
```diff
@@ -139,9 +139,9 @@ export class ParquetEnvelopeReader {
 
     let dictionary;
 
-    const dictionaryPageOffset = colChunk?.meta_data?.dictionary_page_offset;
+    const dictionaryPageOffset = Number(colChunk?.meta_data?.dictionary_page_offset);
```
Comment (ibgreen) [inline modules/parquet/src/parquetjs/classes/parquet-envelope-reader.ts:142]: The test case is called dictionary--offset-zero but we don't read a dictionary when offset zero? 

Can we add more comments to code?

Also it is a little bit suspect to rely on number conversion to convert undefined to 0. Maybe we could be more explicit?

**Review by ibgreen — APPROVED** [review]

File: modules/parquet/src/parquetjs/utils/decoders.ts (hunk 1 above)
Comment (dryabinin-actionengine) [reply modules/parquet/src/parquetjs/utils/decoders.ts:66]: 1. Yes it is defined in 51 line: `let dictionary = options.dictionary || [];`
2. It was done in terms of refactoring parquet module. Need to think about moving it to another folder or do some renaming.

File: modules/parquet/src/parquetjs/classes/parquet-envelope-reader.ts (hunk 2 above)
Comment (dryabinin-actionengine) [reply modules/parquet/src/parquetjs/classes/parquet-envelope-reader.ts:142]: 1. Dictionary with offset 0 file created by` parquet-mr version 1.12.0-201812210311360288-a86293f (build cec1a483e9dcd545e09170ae787d3dcb13744433)` tool. I take it from metadata while debugging process. And it has differences with Impala parquet format in `dictionary_page_offset` and `data_page_offset` fields. It described [here](https://stackoverflow.com/questions/55225108/why-is-dictionary-page-offset-0-for-plain-dictionary-encoding).
Actually it works as it is in our code and I don't fully understand how. The interesting thing that we don't get a dictionary at all but all pages are read as `DATA_PAGE` and has correct values if we compare with` parquet-tools` results for this file.
2. I can describe that can be pages with dictionary offset 0 and it is `parquet-mr` format.
3. Actually yes it is wrong. If `colChunk?.meta_data?.dictionary_page_offset` is `undefined` that means no `dictionaryPage` at all, but if it is 0 looks like it is `parquet- mr` file and we should handle it properly.

Comment (dryabinin-actionengine) [reply modules/parquet/src/parquetjs/classes/parquet-envelope-reader.ts:142]: @ibgreen What I need now is to find out how to get `dictionary` from column chunk in `parquet-mr` files.

**Pull request merged**